tasks/github_tasks.py

```diff
@@ -188,3 +189,60 @@ def get_token_from_app(_, app_id_env='GITHUB_APP_ID', pkey_env='GITHUB_KEY_B64')
     from .libs.common.github_api import GithubAPI
 
     GithubAPI.get_token_from_app(app_id_env, pkey_env)
+
+
+@task
+def assign_team_label(_, pr_id=-1):
+    """
+    Assigns the github team label name if a single team can
+    be deduced from the changed files
+    """
+    from .libs.common.github_api import GithubAPI
+
+    def get_team(changed_files) -> str | None:
+        codeowners = read_owners('.github/CODEOWNERS')
+
+        global_team = None
+        for file in changed_files:
+            owners = [name for (kind, name) in codeowners.of(file) if kind == 'TEAM']
+            if len(owners) != 1:
+                # Multiple / no owners case
+                return
+            else:
+                file_team = owners[0]
+                if global_team is not None and file_team != global_team:
+                    # Multiple owners case
+                    return
+                else:
+                    global_team = file_team
+
+        return global_team
+
+    gh = GithubAPI('DataDog/datadog-agent')
+
+    # Skip if necessary
+    labels = set(gh.get_pr_labels(pr_id))
+
+    if 'qa/done' in labels or 'qa/no-code-change' in labels:
+        print('Qa done or no code change, skipping')
+        return
+
+    if any(label.startswith('team/') for label in labels):
+        print('This PR already has a team label, skipping')
+        return
+
+    # Find team
+    team = get_team(gh.get_pr_files(pr_id))
+    if team is None:
+        print('No team or multiple teams found')
+        return
+    elif not team.startswith('@DataDog/'):
+        print('Invalid team name', team)
+        return
+
+    # Assign label
+    label_name = 'team' + team.removeprefix('@DataDog')
+    if gh.add_pr_label(pr_id, label_name):
```

Comment:
the method `add_pr_label` doesn't return nothing, should we rather encapsulate in a `try: except:`?